riscv-c-api.md

```diff
@@ -164,11 +164,26 @@ For example:
 
 ## Function Attributes
 
-* `__attribute__((naked))`
-* `__attribute__((interrupt))`
-* `__attribute__((interrupt("user")))`
-* `__attribute__((interrupt("supervisor")))`
-* `__attribute__((interrupt("machine")))`
+### `__attribute__((naked))`
+
+The compiler won't generate the prologue/epilogue for those functions with
+`naked` attributes, that usually used when you want to write a function with
+an inline assembly body.
+
+This attribute is incompatible with `interrupt` attribute.
+
+### `__attribute__((interrupt))`, `__attribute__((interrupt("user")))`, `__attribute__((interrupt("supervisor")))` `__attribute__((interrupt("machine")))`
+
+`interrupt` used to specified a function is an interrupt handler. The compiler
+will save/restore all used register at prologue/epilogue regardless of the ABI.
+
+`interrupt` can have an optional parameter for specify the mode, it can be
```

Comment:
The `interrupt` attribute can have an optional parameter to specify the mode.
The possible values are `user`, `supervisor`, or `machine`.
The default value `machine` is used, if the mode is not specified.